rules/valid-expect.js

```diff
@@ -108,6 +142,26 @@ module.exports = {
             }
           }
         }
+
+        if (
+          expectResolvesCase(node) ||
+          expectRejectsCase(node) ||
+          expectNotResolvesCase(node) ||
+          expectNotRejectsCase(node)
+        ) {
+          const parentCallExpressionNode = getParentCallExpressionNode(node);
+          if (!checkIfValidReturn(context, parentCallExpressionNode)) {
+            const { options } = context;
+            const messageReturn =
+              !options[0] || !options[0].alwaysAwait ? ' or returned' : '';
```

Comment:
Extract `!options[0] || !options[0].alwaysAwait` to `allowReturn`? It's used in two places and not very readable

Reply:
(passing just the flag to `checkIfValidReturn` instead of `context`)

Reply:
@jeysal good points, addressed.